packages/dashboard/src/customization/propertiesSections/propertiesAndAlarmsSettings/styledPropertyComponent.tsx

```diff
@@ -153,17 +139,26 @@ export const StyledPropertyComponent: FC<StyledPropertyComponentProps> = ({
     updateStyle(styleToReset); // as we add more sections, reset style values here
   };
 
+  useEffect(() => {
+    if (labelRef.current) {
+      setIsNameTruncated(labelRef.current.scrollWidth > labelRef.current.clientWidth);
+    }
+  }, [label]);
+
   const YAxisHeader = (
     <SpaceBetween size='s' direction='horizontal'>
       {colorable && display === 'property' && (
         <ColorPicker color={property.color || ''} updateColor={(newColor) => updateStyle({ color: newColor })} />
       )}
-      <div style={{ fontWeight: 'normal' }}>{label}</div>
-      <div className='property-display-toggle' onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
-        <Button onClick={onToggleAssetQuery} variant='icon' iconSvg={propertyVisibilityIcon} />
-        <span className='tooltiptext' style={tooltipStyle}>
-          {onMouseOver && isPropertyVisible ? 'hide' : 'unhide'}
-        </span>
+      <Tooltip content={isNameTruncated ? label : ''} position='top'>
```

Comment:
curious: why 2 tooltips ? the screenshot has only one ?

Reply:
tooltip for hide or unhide  is already existing, but here updated to use common tooltip component.
![image](https://github.com/awslabs/iot-app-kit/assets/142866907/5be0e079-b668-42f3-a7e1-36a290b2c845)